java/sage/media/bluray/BluRayFile.java

```diff
@@ -0,0 +1,496 @@
+  {
+    long bytesRead = 0;
+
+    try
+    {
+      ensureProperFile(false, position);
+      bytesRead = sageFileChannel.transferTo(position - fileOffsets[currFileIndex], count, target);
+
+      if (bytesRead < count)
+      {
+        ensureProperFile(false, position);
+        bytesRead += sageFileChannel.transferTo(position + bytesRead - fileOffsets[currFileIndex], count - bytesRead, target);
+      }
+    }
+    finally
+    {
+      ensureProperFile(false);
+    }
+
+    return bytesRead;
+  }
+
+  @Override
+  public long transferTo(long count, WritableByteChannel target) throws IOException
+  {
+    long bytesRead = sageFileChannel.transferTo(count, target);
+
+    if (bytesRead < count)
+    {
+      ensureProperFile(false);
+      bytesRead += sageFileChannel.transferTo(count - bytesRead, target);
+    }
+
+    return bytesRead;
+  }
+
+  @Override
+  public void openFile(String name) throws IOException
+  {
+    sageFileChannel.openFile(name);
+  }
+
+  @Override
+  public void openFile(File file) throws IOException
+  {
+    sageFileChannel.openFile(file);
+  }
+
+  /**
+   * Unsupported method. This channel can only be read only.
+   *
+   * @throws IOException throws channel is read only.
+   */
+  @Override
+  public long transferFrom(ReadableByteChannel src, long position, long count) throws IOException
+  {
+    throw new NonWritableChannelException();
+  }
+
+  /**
+   * Unsupported method. This channel can only be read only.
+   *
+   * @throws IOException throws channel is read only.
+   */
+  @Override
+  public long transferFrom(ReadableByteChannel src, long count) throws IOException
+  {
+    throw new NonWritableChannelException();
+  }
+
+  /**
+   * Unsupported method. This channel can only be read only.
+   *
+   * @throws IOException throws channel is read only.
+   */
+  @Override
+  public int write(ByteBuffer src, long position) throws IOException
+  {
+    throw new NonWritableChannelException();
+  }
+
+  /**
+   * Unsupported method.
+   *
+   * @throws IOException throws method is unsupported.
+   */
+  @Override
+  public MappedByteBuffer map(MapMode mode, long position, long size) throws IOException
+  {
+    throw new IOException("Method is unsupported.");
+  }
+
+  /**
+   * Unsupported method.
+   *
+   * @throws IOException throws method is unsupported.
+   */
+  @Override
+  public FileLock lock(long position, long size, boolean shared) throws IOException
+  {
+    throw new IOException("Method is unsupported.");
+  }
+
+  /**
+   * Unsupported method.
+   *
+   * @throws IOException throws method is unsupported.
+   */
+  @Override
+  public FileLock tryLock(long position, long size, boolean shared) throws IOException
+  {
+    throw new IOException("Method is unsupported.");
+  }
+
+  @Override
+  public int read(ByteBuffer dst, long position) throws IOException
+  {
+    int readBytes = sageFileChannel.read(dst, position - fileOffsets[currFileIndex]);
```

Comment:
I fixed this to work similar to the other methods that provide a position to start reading from.